packages/core/src/browser/tree/tree-model.ts

```diff
@@ -279,11 +279,21 @@ export class TreeModelImpl implements TreeModel, SelectionProvider<ReadonlyArray
     }
 
     protected createBackwardIterator(node: TreeNode | undefined): TreeIterator | undefined {
```

Comment:
Could you elaborate on why these method should be changed?

Reply:
This is moved from https://github.com/theia-ide/theia/blob/master/packages/navigator/src/browser/navigator-model.ts#L146-L172
This is needed for keyboard navigation on filtered nodes